labstep/protocol.py

```diff
@@ -327,6 +329,77 @@ def getSteps(self):
         steps = self.last_version['protocol_steps']
         return listToClass(steps, ProtocolStep, self.__user__)
 
+    def getDataElements(self):
+        """
+        Retrieve the Data Elements of a Protocol.
+
+        Returns
+        -------
+        :class:`~labstep.metadata.Metadata`
+            An array of objects representing the Labstep Data Elements
+            on a Protocol.
+
+        Example
+        -------
+        ::
+
+            protocol = user.getProtocol(17000)
+            metadata = exp_protocol.getDataElements()
+        """
+        metadataThread = self.last_version['metadata_thread']
+        return listToClass(
+            metadataThread['metadatas'],
+            Metadata, self.__user__
+        )
+
+    def addDataElement(self, fieldType="default", fieldName=None,
```

Comment:
This parameter is required by the backend so let's take away the 'None' default to make it required here (you'll also have to move it to be the first param)